### PR TITLE
[MIT-1081] Support billing address for AVS countries

### PR DESCRIPTION
**Pull request by nuxzero**
This PR enhances the credit card form to support billing address for US, UK, and Canadian cardholders.
https://www.omise.co/How-to-improve-my-authorization-rate-for-US-UK-and-Canadian-cardholders

### CODE REVIEW

File: sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt [hunk 1]
```diff
@@ -16,61 +18,145 @@ import co.omise.android.api.RequestListener
...
+                invalidateBillingAddressForm()
+            }
+        }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContentView(R.layout.activity_credit_card)
 
+        require(intent.hasExtra(EXTRA_PKEY)) { "Could not found ${::EXTRA_PKEY.name}." }
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt:130]:
```suggestion
        require(intent.hasExtra(EXTRA_PKEY)) { "Could not find ${::EXTRA_PKEY.name}." }
```

**Review by AnasNaouchi — COMMENTED** [review]
should we have a predefined list of zip codes that is associated to the list of countries or that is not the intended behavior ? @danfowler @nuxzero

**Comment by nuxzero** [comment]
It's not scope of this PR but it's nice to have. Probably we can discuss and make an enhancement in the future.

**Comment by nuxzero** [comment]
> It's not scope of this PR but it's nice to have. Probably we can discuss and make an enhancement in the future.



> should we have a predefined list of zip codes that is associated to the list of countries or that is not the intended behavior ? @danfowler @nuxzero

It's not scope of this PR but it's nice to have. Probably we can discuss and make an enhancement in the future.

**Review by danfowler — APPROVED** [review]

**Review by AnasNaouchi — APPROVED** [review]

File: app/src/java/java/co/omise/android/example/CheckoutActivity.java [hunk 2]
```diff
@@ -240,13 +240,16 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             if (data.hasExtra(OmiseActivity.EXTRA_SOURCE_OBJECT)) {
                 Source source = data.getParcelableExtra(OmiseActivity.EXTRA_SOURCE_OBJECT);
                 snackbar.setText(source.getId()).show();
+                Log.d(TAG, "source: " + source.getId());
```
Comment (danfowler) [inline app/src/java/java/co/omise/android/example/CheckoutActivity.java:243]: We're logging on purpose here for the sake of the example app?

**Review by danfowler — APPROVED** [review]

File: app/src/java/java/co/omise/android/example/CheckoutActivity.java (hunk 2 above)
Comment (nuxzero) [reply app/src/java/java/co/omise/android/example/CheckoutActivity.java:243]: Yes, just in case we want to use the token to create a charge.

**Pull request merged**